docs/source/traits_api_reference/traits.rst

```diff
@@ -21,6 +19,8 @@ Functions
 
 .. autofunction:: Trait
 
+.. autofunction:: Property
```

Comment:
So I guess this is really a `TraitFactory` rather than a `Property`, but it does look as though `autofunction` does more or less the right thing here.

It's unfortunate that the `TraitFactory` wrapper hides the function signature. I wonder whether there's a way around that. (But not in this PR.)